include/envoy/stats/stats.h

```diff
@@ -114,6 +115,11 @@ class Metric {
    * Returns the name of the Metric with the portions designated as tags removed.
    */
   virtual const std::string& tagExtractedName() const PURE;
+
+  /**
+   * Indicates whether a metric has been used.
```

Comment:
this actually is a method used in other Metric as well. Used here signifies whether it has been updated/a value has been set because of some thing happening like connection closing etc. This is helpful in flushing only metrics whose value has been modified. I just moved it to the Metric interface now because histogram also uses it. Earlier Counter and Gauge has this method.   I am not sure if it is equivalent of empty.

Reply:
it is not since last flush, it indicates it has actually ever been set a value.

Reply:
No?  Then what is it used for?  Above you said "flushing only metrics whose value has been modified."

Over time, couldn't you wind up with every metric in this state, if the bit isn't cleared during flush?

Reply:
So it works like this. If a metric value has ever been set it is considered "used" and it would be pushed to stats sinks on subsequent flusher irrespective of whether it has been updated in that interval or not. I think it is reasonable because and would allow only metrics that are really used in that envoy instance to be pushed rather than all metrics. @mattklein123 may have better idea on the history. So I am not changing any thing here.